label_studio/blueprint.py

```diff
@@ -0,0 +1,1085 @@
+    else:
+        project.analytics.send(getframeinfo(currentframe()).function, error=400)
+        return make_response(jsonify("No ML backend"), 400)
+
+
+@blueprint.route('/api/predictions', methods=['POST'])
+@requires_auth
+@exception_treatment
+def api_predictions():
+    """Send creating predictions signal to ML backend"""
+    project = project_get_or_create()
+    if project.ml_backends_connected:
+        # get tasks ids without predictions
+        tasks_with_predictions = {}
+        for task_id, task in project.source_storage.items():
+            task_pred = project.make_predictions(task)
+            tasks_with_predictions[task_pred['id']] = task_pred
+        project.source_storage.set_many(tasks_with_predictions.keys(), tasks_with_predictions.values())
+
+        return make_response(jsonify({'details': 'Predictions done.'}), 200)
+    else:
+        project.analytics.send(getframeinfo(currentframe()).function, error=400)
+        return make_response(jsonify("No ML backend"), 400)
+
+
+@blueprint.route('/version')
+@requires_auth
+@exception_treatment
+def version():
+    """Show backend and frontend version"""
+    lsf = json.load(open(find_dir('static/editor') + '/version.json'))
+    ver = {
+        'label-studio-frontend': lsf,
+        'label-studio-backend': label_studio.__version__
+    }
+    return make_response(jsonify(ver), 200)
+
+
+@blueprint.route('/data/<path:filename>')
+@requires_auth
+@exception_treatment
+def get_data_file(filename):
+    """ External resource serving
+    """
+    project = project_get_or_create()
+
+    # support for upload via GUI
+    if filename.startswith('upload/'):
+        path = os.path.join(project.name, filename)
+        directory = os.path.abspath(os.path.dirname(path))
+        filename = os.path.basename(path)
+        return flask.send_from_directory(directory, filename, as_attachment=True)
+
+    # serving files from local storage
+    if not project.config.get('allow_serving_local_files'):
+        raise FileNotFoundError('Serving local files is not allowed. '
+                                'Use "allow_serving_local_files": true config option to enable local serving')
+    directory = request.args.get('d')
+    return flask.send_from_directory(directory, filename, as_attachment=True)
+
+
+def str2datetime(timestamp_str):
+    try:
+        ts = int(timestamp_str)
+    except:
+        return timestamp_str
+    # return datetime.utcfromtimestamp(ts).strftime('%Y%m%d.%H%M%S')
+    return datetime.utcfromtimestamp(ts).strftime('%c')
+
+
+def start_browser(ls_url, no_browser):
+    import threading
+    import webbrowser
+    if no_browser:
+        return
+
+    browser_url = ls_url + '/welcome'
+    threading.Timer(2.5, lambda: webbrowser.open(browser_url)).start()
+    print('Start browser at URL: ' + browser_url)
+
+
+def main():
+    input_args = parse_input_args()
+    app = create_app(LabelStudioConfig(input_args=input_args), set_str2datetime=True)
```

Comment:
is there a reason that you'd not want the str2datetime filter set? it was previously only set if main ran - so I did this for a neutral refactor